### PR TITLE
dependencies - updated the parent pom to 4.46 and added maven configuration

### PR DESCRIPTION
**Pull request by iqbalrohail**
allow compilation with java 11

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did

### CODE REVIEW

**Comment by iqbalrohail** [comment]
# After mvn clear verify i got these 
Errors ----


[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR] DelegatingDockerClientTest.methodIsDelegatedCorrectly
[ERROR]   Run 1: com.github.dockerjava.api.command.AttachContainerCmd.<init>()
[ERROR]   Run 2: com.github.dockerjava.api.command.AuthCmd.<init>()
[ERROR]   Run 3: 
Mockito cannot mock this class: interface com.github.dockerjava.api.DockerClient.

If you're not sure why you're getting this error, please report to the mailing list.


Java               : 11
JVM vendor name    : Azul Systems, Inc.
JVM vendor version : 11.0.16+8-LTS
JVM name           : OpenJDK 64-Bit Server VM
JVM version        : 11.0.16+8-LTS
JVM info           : mixed mode
OS name            : Linux
OS version         : 5.15.0-46-generic


You are seeing this disclaimer because Mockito is configured to create inlined mocks.
You can learn about inline mocks and their limitations under item #39 of the Mockito class javadoc.

Underlying exception : org.mockito.exceptions.base.MockitoException: Could not modify all classes [interface java.io.Closeable, interface com.github.dockerjava.api.DockerClient, interface java.lang.AutoCloseable]
[ERROR]   Run 4: com.github.dockerjava.api.command.BuildImageCmd.<init>()
[ERROR]   Run 5: java.io.File.<init>()
[ERROR]   Run 6: null
[ERROR]   Run 7: 
Mockito cannot mock this class: interface com.github.dockerjava.api.DockerClient.

If you're not sure why you're getting this error, please report to the mailing list.


Java               : 11
JVM vendor name    : Azul Systems, Inc.
JVM vendor version : 11.0.16+8-LTS
JVM name           : OpenJDK 64-Bit Server VM
JVM version        : 11.0.16+8-LTS
JVM info           : mixed mode
OS name            : Linux
OS version         : 5.15.0-46-generic


You are seeing this disclaimer because Mockito is configured to create inlined mocks.
You can learn about inline mocks and their limitations under item #39 of the Mockito class javadoc.

Underlying exception : org.mockito.exceptions.base.MockitoException: Could not modify all classes [interface java.io.Closeable, interface com.github.dockerjava.api.DockerClient, interface java.lang.AutoCloseable]
[ERROR]   Run 8: com.github.dockerjava.api.command.CommitCmd.<init>()
[ERROR]   Run 9: com.github.dockerjava.api.command.ConnectToNetworkCmd.<init>()
[ERROR]   Run 10: com.github.dockerjava.api.command.ContainerDiffCmd.<init>()
[ERROR]   Run 11: com.github.dockerjava.api.command.CopyArchiveFromContainerCmd.<init>()
[ERROR]   Run 12: com.github.dockerjava.api.command.CopyArchiveToContainerCmd.<init>()
[ERROR]   Run 13: com.github.dockerjava.api.command.CopyFileFromContainerCmd.<init>()
[ERROR]   Run 14: com.github.dockerjava.api.command.CreateConfigCmd.<init>()
[ERROR]   Run 15: com.github.dockerjava.api.command.CreateContainerCmd.<init>()
[ERROR]   Run 16: null
[ERROR]   Run 17: com.github.dockerjava.api.command.CreateNetworkCmd.<init>()
[ERROR]   Run 18: com.github.dockerjava.api.command.CreateSecretCmd.<init>()
[ERROR]   Run 19: com.github.dockerjava.api.command.CreateServiceCmd.<init>()
[ERROR]   Run 20: com.github.dockerjava.api.command.CreateVolumeCmd.<init>()
[ERROR]   Run 21: com.github.dockerjava.api.command.DisconnectFromNetworkCmd.<init>()
[ERROR]   Run 22: com.github.dockerjava.api.command.EventsCmd.<init>()
[ERROR]   Run 23: com.github.dockerjava.api.command.ExecCreateCmd.<init>()
[ERROR]   Run 24: com.github.dockerjava.api.command.ExecStartCmd.<init>()
[ERROR]   Run 25: com.github.dockerjava.api.command.InfoCmd.<init>()
[ERROR]   Run 26: com.github.dockerjava.api.command.InitializeSwarmCmd.<init>()
[ERROR]   Run 27: com.github.dockerjava.api.command.InspectConfigCmd.<init>()
[ERROR]   Run 28: com.github.dockerjava.api.command.InspectContainerCmd.<init>()
[ERROR]   Run 29: com.github.dockerjava.api.command.InspectExecCmd.<init>()
[ERROR]   Run 30: com.github.dockerjava.api.command.InspectImageCmd.<init>()
[ERROR]   Run 31: com.github.dockerjava.api.command.InspectNetworkCmd.<init>()
[ERROR]   Run 32: com.github.dockerjava.api.command.InspectServiceCmd.<init>()
[ERROR]   Run 33: com.github.dockerjava.api.command.InspectSwarmCmd.<init>()
[ERROR]   Run 34: com.github.dockerjava.api.command.InspectVolumeCmd.<init>()
[ERROR]   Run 35: com.github.dockerjava.api.command.JoinSwarmCmd.<init>()
[ERROR]   Run 36: com.github.dockerjava.api.command.KillContainerCmd.<init>()
[ERROR]   Run 37: com.github.dockerjava.api.command.LeaveSwarmCmd.<init>()
[ERROR]   Run 38: com.github.dockerjava.api.command.ListConfigsCmd.<init>()
[ERROR]   Run 39: com.github.dockerjava.api.command.ListContainersCmd.<init>()
[ERROR]   Run 40: com.github.dockerjava.api.command.ListImagesCmd.<init>()
[ERROR]   Run 41: com.github.dockerjava.api.command.ListNetworksCmd.<init>()
[ERROR]   Run 42: com.github.dockerjava.api.command.ListSecretsCmd.<init>()
[ERROR]   Run 43: com.github.dockerjava.api.command.ListServicesCmd.<init>()
[ERROR]   Run 44: com.github.dockerjava.api.command.ListSwarmNodesCmd.<init>()
[ERROR]   Run 45: com.github.dockerjava.api.command.ListTasksCmd.<init>()
[ERROR]   Run 46: com.github.dockerjava.api.command.ListVolumesCmd.<init>()
[ERROR]   Run 47: null
[ERROR]   Run 48: com.github.dockerjava.api.command.LogContainerCmd.<init>()
[ERROR]   Run 49: com.github.dockerjava.api.command.LogSwarmObjectCmd.<init>()
[ERROR]   Run 50: com.github.dockerjava.api.command.LogSwarmObjectCmd.<init>()
[ERROR]   Run 51: com.github.dockerjava.api.command.PauseContainerCmd.<init>()
[ERROR]   Run 52: com.github.dockerjava.api.command.PingCmd.<init>()
[ERROR]   Run 53: com.github.dockerjava.api.command.PruneCmd.<init>()
[ERROR]   Run 54: com.github.dockerjava.api.command.PullImageCmd.<init>()
[ERROR]   Run 55: com.github.dockerjava.api.model.Identifier.<init>()
[ERROR]   Run 56: com.github.dockerjava.api.command.PushImageCmd.<init>()
[ERROR]   Run 57: com.github.dockerjava.api.command.RemoveConfigCmd.<init>()
[ERROR]   Run 58: com.github.dockerjava.api.command.RemoveContainerCmd.<init>()
[ERROR]   Run 59: com.github.dockerjava.api.command.RemoveImageCmd.<init>()
[ERROR]   Run 60: com.github.dockerjava.api.command.RemoveNetworkCmd.<init>()
[ERROR]   Run 61: com.github.dockerjava.api.command.RemoveSecretCmd.<init>()
[ERROR]   Run 62: com.github.dockerjava.api.command.RemoveServiceCmd.<init>()
[ERROR]   Run 63: com.github.dockerjava.api.command.RemoveSwarmNodeCmd.<init>()
[ERROR]   Run 64: com.github.dockerjava.api.command.RemoveVolumeCmd.<init>()
[ERROR]   Run 65: com.github.dockerjava.api.command.RenameContainerCmd.<init>()
[ERROR]   Run 66: com.github.dockerjava.api.command.ResizeContainerCmd.<init>()
[ERROR]   Run 67: com.github.dockerjava.api.command.ResizeExecCmd.<init>()
[ERROR]   Run 68: com.github.dockerjava.api.command.RestartContainerCmd.<init>()
[ERROR]   Run 69: com.github.dockerjava.api.command.SaveImageCmd.<init>()
[ERROR]   Run 70: com.github.dockerjava.api.command.SaveImagesCmd.<init>()
[ERROR]   Run 71: com.github.dockerjava.api.command.SearchImagesCmd.<init>()
[ERROR]   Run 72: com.github.dockerjava.api.command.StartContainerCmd.<init>()
[ERROR]   Run 73: com.github.dockerjava.api.command.StatsCmd.<init>()
[ERROR]   Run 74: com.github.dockerjava.api.command.StopContainerCmd.<init>()
[ERROR]   Run 75: com.github.dockerjava.api.command.TagImageCmd.<init>()
[ERROR]   Run 76: com.github.dockerjava.api.command.TopContainerCmd.<init>()
[ERROR]   Run 77: com.github.dockerjava.api.command.UnpauseContainerCmd.<init>()
[ERROR]   Run 78: com.github.dockerjava.api.command.UpdateContainerCmd.<init>()
[ERROR]   Run 79: com.github.dockerjava.api.command.UpdateServiceCmd.<init>()
[ERROR]   Run 80: com.github.dockerjava.api.command.UpdateSwarmCmd.<init>()
[ERROR]   Run 81: com.github.dockerjava.api.command.UpdateSwarmNodeCmd.<init>()
[ERROR]   Run 82: com.github.dockerjava.api.command.VersionCmd.<init>()
[ERROR]   Run 83: com.github.dockerjava.api.command.WaitContainerCmd.<init>()
[INFO] 
[INFO] 
[ERROR] Tests run: 5, Failures: 0, Errors: 1, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  46.111 s
[INFO] Finished at: 2022-08-21T21:06:00Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M7:test (default-test) on project docker-java-api: 
[ERROR] 
[ERROR] Please refer to /workspace/docker-java-api-plugin/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

File: pom.xml [hunk 1]
```diff
@@ -146,11 +158,13 @@
...
-            <version>2.10.0</version>
-            <scope>test</scope>
-        </dependency>
+        <!-- https://mvnrepository.com/artifact/org.mockito/mockito-core -->
+<dependency>
+    <groupId>org.mockito</groupId>
+    <artifactId>mockito-core</artifactId>
+    <version>4.7.0</version>
+    <scope>test</scope>
+</dependency>
```
Comment (dheerajodha) [inline pom.xml:167]: Please fix the indentation.

File: pom.xml [hunk 2]
```diff
@@ -26,12 +25,25 @@
...
+        <dependencies>
+          <dependency>
+            <groupId>io.jenkins.tools.bom</groupId>
+            <artifactId>bom-2.332.x</artifactId>
+            <version>1595.v8c71c13cc3a_9</version>
+            <type>pom</type>
+            <scope>import</scope>
+          </dependency>
+        </dependencies>
+      </dependencyManagement>
```
Comment (dheerajodha) [inline pom.xml:45]: Fix the indentation here as well.

File: pom.xml [hunk 3]
```diff
@@ -26,12 +25,25 @@
     <properties>
         <revision>3.2.13</revision>
         <changelist>999999-SNAPSHOT</changelist>
-        <jenkins.version>1.609.1</jenkins.version>
+       <jenkins.version>2.332.4</jenkins.version> <!-- used updated version of jenkins -->
```
Comment (dheerajodha) [inline pom.xml:28]: Please fix the indentation here as well.

**Review by dheerajodha — COMMENTED** [review]
Thanks for the PR, as far as the content of this PR is concerned, I'd wait for the plugin maintainer to comment.

I just wanted to share my observation on the improper indentation in your changes. Please take care of it.

File: pom.xml (hunk 2 above)
Comment (iqbalrohail) [reply pom.xml:45]: fixed

File: pom.xml (hunk 3 above)
Comment (iqbalrohail) [reply pom.xml:28]: fixed

File: pom.xml (hunk 1 above)
Comment (iqbalrohail) [reply pom.xml:167]: fixed

File: pom.xml [hunk 4]
```diff
@@ -146,11 +158,13 @@
...
         </dependency>
+        <!-- https://mvnrepository.com/artifact/org.mockito/mockito-core -->
         <dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-core</artifactId>
-            <version>2.10.0</version>
+            <version>4.7.0</version>
             <scope>test</scope>
         </dependency>
+
```
Comment (dheerajodha) [inline pom.xml:168]: I don't think there's a need for this new line here. Please look at lines 67 and 68, they do not have any space between them.

**Comment by ericcitaire** [comment]
Superseded by https://github.com/jenkinsci/docker-java-api-plugin/pull/56

**Pull request closed without merging**